data.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Holds the data associated with a website. Websites are identified by origin.
+ */
+class AkitaOriginData {
+  origin = null;
+  isMonetized = false;
+  // Object<AkitaPaymentPointerData>
+  paymentPointerData = {};
+
+  constructor(origin) {
+    this.origin = origin;
+  }
+
+  static fromObject(akitaOriginDataObject) {
+    const newThis = new AkitaOriginData(akitaOriginDataObject.origin);
+    newThis.isMonetized = akitaOriginDataObject.isMonetized;
+
+    for (const paymentPointer in akitaOriginDataObject.paymentPointerData) {
+      newThis.paymentPointerData[paymentPointer] = AkitaPaymentPointerData.fromObject(
+        akitaOriginDataObject.paymentPointerData[paymentPointer]
+      );
+    }
+
+    return newThis;
+  }
+
+  /**
+   *
+   * @param {{
+   *  paymentPointer: string,
+   *  assetCode?: string,
+   *  assetScale?: number,
+   *  amount?: number
+   * }} monetizationObject
+   *   This object may be created, or a Web Monetization event detail object can be used.
+   *   Pass in an object with just a paymentPointer to register a payment pointer for
+   *   the current website. Payment pointer should be validated first.
+   *   Additionally pass in assetCode, assetScale, and amount to add to the
+   *   total amount sent to the current website.
+   */
+  addMonetization({
+    paymentPointer,
+    assetCode = null,
+    assetScale = null,
+    amount = null
+  }) {
+    if (paymentPointer) {
+      this.isMonetized = true;
+      if (!this.paymentPointerData[paymentPointer]) {
+        this.paymentPointerData[paymentPointer] = new AkitaPaymentPointerData(paymentPointer);
+      }
+      if (assetCode && amount && assetScale !== null) {
+        this.paymentPointerData[paymentPointer].addAsset(assetCode, Number(amount), Number(assetScale));
+      }
+    }
+
+    return this;
+  }
+}
+
+class AkitaPaymentPointerData {
+  paymentPointer = null;
+  // Object<WebMonetizationAsset>
```

Comment:
is this needed?

Reply:
No, just annotating the type to try to make it more clear how the data is structured. Modified the comment so it is hopefully more clear what it means.